overlord/ifacestate/hotplug/export_test.go

```diff
@@ -0,0 +1,32 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package hotplug
+
+import (
+	"github.com/snapcore/snapd/osutil/udev/netlink"
+)
+
+func MockUDevMonitorChannel(mon *UDevMonitor, events chan netlink.UEvent) {
+	mon.netlinkEvents = events
+}
+
+func MockUDevMonitorStopChannel(mon *UDevMonitor, monitorStop chan struct{}) {
```

Comment:
The stop channel is normally created and returned by `go-udev` implementation of its `Monitor` when created, and it's the way for the caller to stop Monitor's goroutine.
In the udevmon_test (not visible here, already in master) I cannot run the real go-udev monitor and instead mock both its channels (stop channel and event channel; the latter is used in the tests to inject fake events). `go-udev` doesn't use tombs.

Reply:
I still find that undesired. We shouldn't have to dig into go-udev's implementation from our tests like this, even more when it's about replacing a channel. The most important reason why people use channels like this is to avoid races, but if you replace the channel you've got a race.

Let's please discuss this in the standup or in a specific call.